lib/rack/builder.rb

```diff
@@ -35,39 +35,6 @@ class Builder
     # https://stackoverflow.com/questions/2223882/whats-the-difference-between-utf-8-and-utf-8-without-bom
     UTF_8_BOM = '\xef\xbb\xbf'
 
-    # Parse the given config file to get a Rack application.
-    #
-    # If the config file ends in +.ru+, it is treated as a
-    # rackup file and the contents will be treated as if
-    # specified inside a Rack::Builder block.
-    #
-    # If the config file does not end in +.ru+, it is
-    # required and Rack will use the basename of the file
-    # to guess which constant will be the Rack application to run.
-    #
-    # Examples:
-    #
-    #   Rack::Builder.parse_file('config.ru')
-    #   # Rack application built using Rack::Builder.new
-    #
-    #   Rack::Builder.parse_file('app.rb')
-    #   # requires app.rb, which can be anywhere in Ruby's
-    #   # load path. After requiring, assumes App constant
-    #   # contains Rack application
-    #
-    #   Rack::Builder.parse_file('./my_app.rb')
-    #   # requires ./my_app.rb, which should be in the
-    #   # process's current directory.  After requiring,
-    #   # assumes MyApp constant contains Rack application
-    def self.parse_file(path)
```

Comment:
This removes the ability for rackup to take an `.rb` file instead of an `.ru`, which breaks backwards compatibility.  It seems unrelated to `rack.multithread`/`rack.multiprocess`/`rack.run_once`.  The entire commit that removes this method seems unrelated to `rack.multithread`/`rack.multiprocess`/`rack.run_once`, and I don't think should be considered in this pull request.

Reply:
It's related in the sense that the replacement (Builder#options) cannot possibly make sense with a static middleware loaded from a `.rb` file. If we are going to continue to support this, it would make more sense for us to invoke `#call` on the class and pass in a builder instance. But we can have this in another PR, it just means there will be a point where the master branch no longer exposes any concurrency/parallelism metadata. This PR was to remove it and also propose a replacement.